wumpus.py

```diff
@@ -103,7 +107,99 @@ def _breeze_stench_rules(self):
         return clauses
 
 
-def create_wumpus_world():
+class WumpusWorldGenerator:
+    def __init__(self, map_width=4, map_height=4):
+        self._map_width = map_width
+        self._map_height = map_height
+        self._occupied_spaces = []
+        self._generate_random_wumpus()
+
+
+    def _spawn_object_coords(self):
+        grid_size = self._map_width * self._map_height
+        acceptable_range = range(grid_size) # Nothing should be on 0,0
+        if len(self._occupied_spaces) > 0:
```

Comment:
What does this do? It breaks the code as a range object has no `remove` method. I tried removing that line and got:

```
[set(), set(), {<Property.GOLD: 2>, <Property.STENCH: 1>}, set()]
[{<Property.GOLD: 2>}, {<Property.GOLD: 2>, <Property.STENCH: 1>}, {<Property.WUMPUS: 4>, <Property.GOLD: 2>}, {<Property.GOLD: 2>, <Property.STENCH: 1>}]
[set(), set(), {<Property.STENCH: 1>}, {<Property.GOLD: 2>}]
[{<Property.GOLD: 2>}, {<Property.GOLD: 2>}, set(), set()]
```

To make things simpler, just traverse again the grid after you have filled the cave with PITs and wumpus. Get the safe cells and pick one at random and put the gold there

Reply:
ah my bad, it should be a list(range(grid_size)). basically trying to ravel the grid indices, remove the occupied spaces from the list of possible locations, randomly choose one of the remaining spaces, and then re-convert that into x,y coordinates

Reply:
figured out why no traps were spawning: https://github.com/dpalmasan/wumpus-ai/pull/27/commits/63c2e34b79bd21e85a98b03c55d89f7f02770185